### PR TITLE
Add test suite

### PR DESCRIPTION
**Pull request by jonandernovella**
An initial test suite has been implemented, which is tested against the development environment.

Fixes #22

### CODE REVIEW

**Comment by jonandernovella** [comment]
> This PR contains so muck more than the test suite that is should be split into at least 2 PRs, one for the tests and the one or more for the code changes.

The thing is that the base branch is the one from my previous PR. There the reason for so many things showing up.

File: dev-server/oidc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM node:10
```
Comment (jbygdell) [inline dev-server/oidc/Dockerfile]:
```suggestion
FROM node:lts-slim
```
Even if it's only used in a CI environment, try to use the smallest possible image. It does take longer to download 350MB image over of a 50MB image.
It's also preferable to use a LTS release unless there are uncontrollable requirements that forces the use of an older image.

File: dev-server/oidc/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "mock-oidc",
+  "version": "0.1.0",
+  "main": "server.js",
+  "engines": {
+    "node": ">=10.15.0 <12"
```
Comment (jbygdell) [inline dev-server/oidc/package.json]: This feels a bit outdated, the current LTS relase is 12. This forces the use of an older version.
```suggestion
    "node": ">=12"
```

**Comment by jbygdell** [comment]
Commit c31d74b through 3abdcf9 should not really be part of this PR since they are in PR #21

**Comment by viklund** [comment]
> Commit [c31d74b](https://github.com/NBISweden/sda-auth/commit/c31d74bcae32df555c52e33f065f002f13fc2b61) through [3abdcf9](https://github.com/NBISweden/sda-auth/commit/3abdcf9a03a5796a80d2128a8f0c9b1fb75a2d46) should not really be part of this PR since they are in PR #21

Correct. Once #21 is merged this will be rebased.

**Pull request merged**